kratos/processes/from_json_check_result_process.cpp

```diff
@@ -0,0 +1,866 @@
+
+                    const double result = it_node->GetValue(*p_var_double);
+                    const double reference = r_var_database.GetValue(i, time);
+                    if (!CheckValues(result, reference)) {
+                        FailMessage(it_node->Id(), "Node", result, reference, p_var_double->Name());
+                        check_counter += 1;
+                    }
+                }
+            }
+            for (auto& p_var_array : mpNodalVariableArrayList) {
+                const auto& r_var_database = r_node_database.GetVariableData(*p_var_array);
+
+                #pragma omp parallel for reduction(+:check_counter)
+                for (int i = 0; i < static_cast<int>(r_nodes_array.size()); ++i) {
+                    auto it_node = it_node_begin + i;
+
+                    const auto& r_entity_database = r_var_database.GetEntityData(i);
+                    const array_1d<double, 3>& r_result = it_node->GetValue(*p_var_array);
+                    for (IndexType i_comp = 0; i_comp < 3; ++i_comp) {
+                        const double reference = r_entity_database.GetValue(time, i_comp);
+                        if (!CheckValues(r_result[i_comp], reference)) {
+                            FailMessage(it_node->Id(), "Node", r_result[i_comp], reference, p_var_array->Name());
+                            check_counter += 1;
+                        }
+                    }
+                }
+            }
+            for (auto& p_var_vector : mpNodalVariableVectorList) {
+                const auto& r_var_database = r_node_database.GetVariableData(*p_var_vector);
+
+                #pragma omp parallel for reduction(+:check_counter)
+                for (int i = 0; i < static_cast<int>(r_nodes_array.size()); ++i) {
+                    auto it_node = it_node_begin + i;
+
+                    const auto& r_entity_database = r_var_database.GetEntityData(i);
+                    const Vector& r_result = it_node->GetValue(*p_var_vector);
+                    for (IndexType i_comp = 0; i_comp < r_result.size(); ++i_comp) {
+                        const double reference = r_entity_database.GetValue(time, i_comp);
+                        if (!CheckValues(r_result[i_comp], reference)) {
+                            FailMessage(it_node->Id(), "Node", r_result[i_comp], reference, p_var_vector->Name());
+                            check_counter += 1;
+                        }
+                    }
+                }
+            }
+        }
+
+        // Getting process info
+        const auto& r_process_info = mrModelPart.GetProcessInfo();
+
+        // Iterate over elements
+        const auto& r_elements_array = GetElements();
+        const auto it_elem_begin = r_elements_array.begin();
+
+        // Result vectors
+        std::vector<double> result_double;
+        std::vector<array_1d<double,3>> result_array;
+        std::vector<Vector> result_vector;
+
+        for (auto& p_var_double : mpGPVariableDoubleList) {
```

Comment:
Are you getting a reference to a pointer in here?

Reply:
Pointers